### DISCUSSION TITLE
Upgrade to Jakarta EE 8

### DISCUSSION THREAD

**Discussion by erikgb** [eclipse-rdf4j/rdf4j#4433] [Ideas]
As an intermediate step towards migrating from Java EE to Jakarta EE, I suggest starting with an upgrade to Jakarta EE 8 - which is supposed to be fully compatible with Java EE 8. This is the initial Jakarta EE release - before the troublesome javax->jakarta namespace change.

I have created a WIP PR for this, https://github.com/eclipse/rdf4j/pull/4397, and it seems like the current blocker is WireMock - where a release supporting Java/Jakarta EE 8 does not exists, and there are no plans to fix this, ref. https://github.com/wiremock/wiremock/issues/1967.

I have created an issue, based on a comment from @abrokenjester, https://github.com/eclipse/rdf4j/issues/4431, and hope to get some consensus on the direction forward from the maintainers. 😃

**Comment by erikgb** [comment]
After fixing the WireMock issues in https://github.com/eclipse/rdf4j/pull/4439, the current blocker is the required Jetty upgrade to version 10 (ref. https://github.com/eclipse/rdf4j/issues/4252). Upgrading Jetty in our codebase seems trivial to do, but Solr is also using Jetty, triggering a chain of upgrades. I've created https://github.com/eclipse/rdf4j/issues/4441 for now, but after the comment from @barthanssens I wonder if there is another way of approaching this "knot"?

WDYT about removing all Jetty dependencies from our codebase and migrating to JUnit Server fueled by Tomcat? Ref. https://github.com/eclipse/rdf4j/pull/4437 (POC PR using Jetty underneath, but JUnit Server also supports Tomcat).

Or do you see another way of approaching this? @abrokenjester @hmottestad

**Reply by erikgb** [reply]
Upgrading Solr seems to be a dead end, as the latest version still [depends on Jetty 9](https://central.sonatype.com/artifact/org.apache.solr/solr-core/9.1.1/dependencies). 😢 Do you think it is possible to eliminate the dependency to solr-core somehow? It seems to be required to support running Solr embedded - but that is not recommended by the Solr maintainers, ref. https://cwiki.apache.org/confluence/display/solr/EmbeddedSolr. That would eventually make the solr-sail module only depend on EOL **client** modules.

**Reply by barthanssens** [reply]
IMHO upgrading is not really a dead end per se, but rather something that should be part of a 5.0 release (since it also involves newer Lucene libraries and maybe some other dependencies as well...)

**Reply by erikgb** [reply]
But that will require an upgrade upstream (Solr), or am I wrong? Could you elaborate your idea, @barthanssens?

**Reply by barthanssens** [reply]
If all goes well, an upcoming Solr 9.2 would move to Jetty 10: https://issues.apache.org/jira/browse/SOLR-15955
